packages/vscode-extension/src/connectionService.ts

```diff
@@ -286,15 +288,62 @@ export async function establishPersistentConnectionToSchemaPoller(
   );
 
   result.success
-    ? attachSchemaPollerConnectionErrorEventListener()
+    ? attachSchemaPollerConnectionEventListeners()
     : attachSchemaPollerConnectionFailedEventListeners();
 
   return result;
 }
 
-export function getConnectionDatabases(): Database[] {
+/**
+ *  Gets an array of database names from the connection, if they exist.
+ * @returns An array of database names, or an empty array if no databases exist.
+ */
+export function getConnectionDatabases(): Pick<
+  Database,
+  'name' | 'default' | 'home'
+>[] {
+  const schemaPoller = getSchemaPoller();
+  const databases = schemaPoller.connection?.databases ?? [];
+
+  if (
+    !schemaPoller.metadata ||
+    !schemaPoller.metadata.dbSchema?.databaseNames
+  ) {
+    return databases;
+  }
+
+  return schemaPoller.metadata.dbSchema.databaseNames.map((name) => {
+    const database = databases.find((db) => db.name === name);
+    if (!database) {
+      return {
+        name: name,
+        default: false,
+        home: false,
+      };
+    } else {
+      return database;
+    }
+  }, []);
+}
+
+/**
+ * Returns the labels and relationship types from the dbSchema, if they exist.
+ * @returns An object containing the labels and relationships, or null if the dbSchema does not exist.
+ */
+export function getDbSchemaInformation(): {
+  labels: string[];
+  relationships: string[];
+} | null {
   const schemaPoller = getSchemaPoller();
-  return schemaPoller.connection?.databases ?? [];
+
+  if (!schemaPoller.metadata || !schemaPoller.metadata.dbSchema) {
+    return null;
```

Comment:
Suggestion to make this conformant to how most of the code of the repo is written: normally we return undefined instead of null

Reply:
good shout :)